src/clr-core/button/button.element.scss

```diff
@@ -1,35 +1,55 @@
+// Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+// This software is released under MIT license.
+// The full license information can be found in LICENSE in the root directory of this project.
+
+@import './../styles/tokens/generated/index';
+
 @import './../styles/mixins/utils';
+@import './../styles/mixins/mixins';
 
-// TODO: progress spinner CSS deprecated, will be replaced by cds-progress
+// TODO: progress spinner CSS deprecated, will be replaced by cds-circular-progress
 @import './../../clr-angular/progress/spinner/variables.spinner';
 @import './../../clr-angular/image/icons.clarity';
 @import './../../clr-angular/progress/spinner/spinner.clarity';
 
 :host {
-  --box-shadow-color: var(--clr-color-action-500, #{$clr-color-action-500});
-  --border-radius: #{$clr_baselineRem_3px};
-  --border-color: var(--clr-color-action-600, #{$clr-color-action-600});
-  --color: var(--clr-color-neutral-0, #{$clr-color-neutral-0});
-  --background: var(--clr-color-action-600, #{$clr-color-action-600});
+  --box-shadow-color: #{$cds-token-color-action-900};
+  --border-radius: #{$cds-token-global-border-radius};
+  --border-width: #{$cds-token-global-border-width};
+  --border-color: #{$cds-token-color-action-600};
+  --color: #{$cds-token-color-neutral-0};
+  --background: #{$cds-token-color-action-600};
+  --height: #{$clr_baselineRem_1_5};
   display: inline-block;
+  -webkit-appearance: none !important;
+}
+
+@mixin button-padding() {
+  padding-top: calc(var(--padding-vertical) - var(--border-width) + 0.025rem);
+  padding-left: calc(var(--padding-horizontal) - var(--border-width));
+  padding-right: calc(var(--padding-horizontal) - var(--border-width));
+  padding-bottom: calc(var(--padding-vertical) - var(--border-width) - 0.025rem);
```

Comment:
Why are the calcs necessary as well as the 0.025rem?

Reply:
For specific sizing, we have to remove the border-width from the padding values. Otherwise, we get a 37.5px high button instead of the expected 36px.

The `0.025rem` is a magic number that nudges the button text just enough to make it perfectly centered along the vertical axis.

Reply:
I'm revisiting the LHEs, tho. So if I can get that working for buttons, we may be able to get rid of this magic number.